File: palindrome_simple/palindrome_str.cpp
#include <stdio.h>
#include <iostream>
#include <math.h>
#include <string>

using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        string num = to_string(x);
        int len = num.size();
        int cntr = int(len/2);
        for(int i=0; i < int(len/2); i++){
            if (num[i] == num[len-i-1]){
                --cntr;
            }
        }
        if (cntr == 0){
            return true;
        }
        return false;
    }
};

File: palindrome_simple/palindrome_str.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        str_x = str(x)
        str_len = len(str_x)
        cntr = str_len//2
        
        for i in range(str_len//2):
            if str_x[i] == str_x[str_len-i-1]:
                cntr -= 1
        if cntr == 0:
            return True
        return False